addons/account/models/account_bank_statement.py

```diff
@@ -1024,6 +1024,10 @@ def _prepare_reconciliation(self, lines_vals_list, create_payment_for_invoice=Fa
         '''
 
         self.ensure_one()
+        journal = self.journal_id
+        company_currency = journal.company_id.currency_id
+        journal_currency = journal.currency_id or company_currency
+        foreign_currency = self.foreign_currency_id or journal_currency or company_currency
```

Comment:
self.foreign_currency_id or journal_currency would be enough; journal currency is already doing "or company_currency"
==> You could as well just not define any journal_currency variable (it's not used) , and simply do

foreign_currency =  self.foreign_currency_id or journal.currency_id or company_currency